agent/node.go

```diff
@@ -271,19 +278,17 @@ func (n *Node) run(ctx context.Context) (err error) {
 		for {
 			select {
 			case certUpdate := <-updates:
-				if ctx.Err() != nil {
-					return
-				}
 				if certUpdate.Err != nil {
 					logrus.Warnf("error renewing TLS certificate: %v", certUpdate.Err)
 					continue
 				}
-				n.Lock()
-				n.role = certUpdate.Role
-				if n.role == ca.ManagerRole {
-					n.managerRoleCh <- struct{}{}
+				// switch role to manager only when we got cert
+				if certUpdate.Role == ca.ManagerRole {
```

Comment:
Why do we need this check in here? I know usually switching to worker should happen in `roleChangeReq` but if in some rare condition we get a switch to worker here shouldn't we still set it? Otherwise, we can try to start manager using a worker cert.